File: Rnwood.Smtp4dev/DbModel/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity.Core.Objects.DataClasses;

namespace Rnwood.Smtp4dev.DbModel
{
    public class Message
    {
        [Key] public Guid Id { get; set; }

        public long ImapUid { get; internal set; }

        public string From { get; set; }
        public string To { get; set; }
        public DateTime ReceivedDate { get; set; }

        public string Subject { get; set; }

        public byte[] Data { get; set; }

        public string MimeParseError { get; set; }
        
        public string SessionEncoding { get; set; }

        public Session Session { get; set; }

        public Mailbox Mailbox { get; set; }
        public Guid? MailboxFolderId { get; set; }
        public MailboxFolder MailboxFolder { get; set; }
        public int AttachmentCount { get; set; }

        public bool IsUnread { get; set; }
        public string RelayError { get; internal set; }

        public bool SecureConnection { get; set; }
        
        public bool? EightBitTransport { get; set; }
        
        public bool HasBareLineFeed { get; set; }

        /// <summary>
        /// JSON metadata extracted from MIME message including CC recipients, attachment filenames, etc.
        /// </summary>
        public string MimeMetadata { get; set; }
        
        /// <summary>
        /// Plain text body content extracted from MIME message for searching
        /// </summary>
        public string BodyText { get; set; }

        public virtual List<MessageRelay> Relays { get; set; } = new List<MessageRelay>();
        public string DeliveredTo { get; set; }

        public void AddRelay(MessageRelay messageRelay)
        {
            messageRelay.Message = this;
            this.Relays.Add(messageRelay);
        }
    }
}

File: Rnwood.Smtp4dev/ClientApp/src/ApiClient/Message.ts
//MessageEntitySummary[] from Message
import MessageEntitySummary from "./MessageEntitySummary";
//Header[] from Message 
import Header from "./Header";
export default class Message {
  constructor(
    id: string,
    from: string,
    to: string,
    cc: string,
    bcc: string,
    receivedDate: Date,
    subject: string,
    parts: MessageEntitySummary[],
    headers: Header[],
    mimeParseError: string,
    relayError: string,
    secureConnection: boolean,
    hasHtmlBody: boolean,
    hasPlainTextBody: boolean
  ) {
    this.id = id;
    this.from = from;
    this.to = to;
    this.cc = cc;
    this.bcc = bcc;
    this.receivedDate = receivedDate;
    this.subject = subject;
    this.parts = parts;
    this.headers = headers;
    this.mimeParseError = mimeParseError;
    this.relayError = relayError;
    this.secureConnection = secureConnection;
    this.hasHtmlBody = hasHtmlBody;
    this.hasPlainTextBody = hasPlainTextBody;
  }

  id: string;
  from: string;
  to: string;
  cc: string;
  bcc: string;
  receivedDate: Date;
  subject: string;
  parts: MessageEntitySummary[];
  headers: Header[];
  mimeParseError: string;
  relayError: string;
  secureConnection: boolean;
  hasHtmlBody: boolean;
  hasPlainTextBody: boolean;
}
